BrightID/src/components/Connections/ConnectionScreenController.js

```diff
@@ -25,35 +25,38 @@ function ConnectionScreenController(props: ConnectionScreenProps) {
   );
   const myConnections = useSelector((state) => state.connections.connections);
   const myGroups = useSelector((state) => state.groups.groups);
+  const [connectionsNum, setConnectionsNum] = useState(0);
+  const [groupsNum, setGroupsNum] = useState(0);
   const [mutualGroups, setMutualGroups] = useState<Array<group>>([]);
-  const [mutualConnections, setMutualConnections] = useState<Array<connection>>(
-    [],
-  );
-  const [verifications, setVerifications] = useState<Array<string>>([]);
+  const [mutualConnections, setMutualConnections] = useState<Array<connection>>([]);
+  const [verified, setVerified] = useState(false);
+  const [createdAt, setCreatedAt] = useState(0);
+  const [connectedAt, setConnectedAt] = useState(0);
   const [loading, setLoading] = useState(false);
 
   useFocusEffect(
     useCallback(() => {
       const fetchData = async (connectionId) => {
         setLoading(true);
         console.log(`fetching connection info for ${connectionId}`);
-        const connectionData = await fetchConnectionInfo({
-          brightId: connectionId,
-          myConnections,
-          myGroups,
+        const profile = await api.getUserProfile(connectionId);
+        console.log(profile, 22);
+        setConnectionsNum(profile.connectionsNum);
+        setGroupsNum(profile.groupsNum);
+        setVerified(profile.verified);
+        setCreatedAt(profile.createdAt);
+        setConnectedAt(profile.connectedAt);
+        profile.mutualConnections = myConnections.filter(function (conn) {
+          return profile.mutualConnections.includes(conn.id);
+        });
+        setMutualConnections(profile.mutualConnections);
+        profile.mutualGroups = myGroups.filter(function (group) {
+          return profile.mutualGroups.includes(group.id);
         });
-        setMutualGroups(connectionData.mutualGroups);
-        setMutualConnections(connectionData.mutualConnections);
-        setVerifications(connectionData.verifications);
+        setMutualGroups(profile.mutualGroups);
         setLoading(false);
       };
-      if (connection) {
```

Comment:
Maybe it's better to keep/adjust this check (`if connectionId !== undefined`), to prevent unnecessary backend calls with undefined connectionId? I'm not sure that can happen, but it should not harm either :)